bftengine/src/bcstatetransfer/BCStateTran.hpp

```diff
@@ -80,7 +80,11 @@ class BCStateTran : public IStateTransfer {
   void handleStateTransferMessage(char* msg, uint32_t msgLen, uint16_t senderId) override;
 
  protected:
-  const bool pedanticChecks_;
+  std::function<void(char*, uint32_t, uint16_t)> messageHandler_;
+  // actual handling function. can be used in context of dedicated thread
+  void handleStateTransferMessageImp(char* msg, uint32_t msgLen, uint16_t senderId);
+  // handling from other context
+  void handoff(char* msg, uint32_t msgLen, uint16_t senderId);
```

Comment:
Can we make these functions accept a `const void*` buffer instead?

Reply:
Basically yes, but we'll need to replace all classes implementing `IStateTransfer::handleStateTransferMessage`
If you think it's essential, let's do it as a separate effort.

Reply:
Sure, separate one is fine.